src/data/finset/n_ary.lean

```diff
@@ -0,0 +1,238 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.finset.prod
+
+/-!
+### N-ary images of finsets
+
+This file defines the binary image of finsets. This is mostly useful to define pointwise operations
+on finsets.
+
+## Notes
+
+This file is very similar to the n-ary section of `data.set.basic` and to `order.filter.n_ary`.
+Please keep them in sync.
+
+We do not define `finset.image₃` as its only purpose would be to prove properties of `finset.image₂`
+and `set.image2` already fulfills this task.
+-/
+
+open function set
```

Comment:
How many lemmas actually use this `set`? I can see us renaming `set.image2` to `set.image₂` in the near future, and then this is just going to cause pain. I'd lean towards
```suggestion
open function
```

Reply:
I've just tried it out, by renaming `finset.image₂` into `finset.image2`. There's literally only one place where I need to add `set.` (the other way around requires 15 `set.`). The reason is that when we call `whatever`, it's to prove `finset.whatever` so the name can only be resolved to `set.whatever`.